laravel/app/Http/Controllers/InstanceController.php

```diff
@@ -32,15 +32,10 @@ public function overview(): View
                 $virtualserver_helper = new TeamSpeakVirtualserver($instance);
                 $virtualserver = $virtualserver_helper->get_virtualserver_connection();
                 $channel_list = $virtualserver->channelList();
+                $channelListForEachInstance[$instance->id] = $channel_list;
```

Comment:
[VETO]

You're now sometimes saving a `ChannelList` into this array and sometimes an `ExceptionMessage`.

I would recommend to save channel list to a dedicated key - like you do it for the `error`:
```suggestion
                $channelListForEachInstance[$instance->id]['channel_list'] = $virtualserver->channelList();;
```

And in case of an exception / error, you could simply set the channel list to an empty array:
```php
                $channelListForEachInstance[$instance->id]['channel_list'] = [];
```

Then it's clear in the code, that there are no channels to display and with the additional `error` key you can determine in the view to either display an error message or not.